src/LiveComponent/src/Util/QueryStringPropsExtractor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Util;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\UX\LiveComponent\Exception\HydrationException;
+use Symfony\UX\LiveComponent\LiveComponentHydrator;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadata;
+use Symfony\UX\LiveComponent\Metadata\LivePropMetadata;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+final class QueryStringPropsExtractor
+{
+    public function __construct(private readonly LiveComponentHydrator $hydrator)
+    {
+    }
+
+    /**
+     * Extracts relevant query parameters from the current URL and hydrates them.
+     */
+    public function extract(Request $request, LiveComponentMetadata $metadata, object $component): array
+    {
+        $query = $request->query->all();
+
+        if (empty($query)) {
+            return [];
+        }
+        $data = [];
+
+        foreach ($metadata->getAllLivePropsMetadata() as $livePropMetadata) {
+            if ($queryStringMapping = $livePropMetadata->getQueryStringMapping()) {
+                if (null !== ($value = $query[$queryStringMapping['name']] ?? null)) {
+                    if (\is_array($value) && $this->isNumericIndexedArray($value)) {
+                        // Sort numeric array
```

Comment:
Why ?

Reply:
Originally, it was because `InputBag::all()` (used to get query params) doesn't sort array values even if the array contains only numeric keys. The order of elements is the same as their order in the URL. So it depends on how the URL is modified on client side. 

Now the JS code writes array elements in the correct order, so it's not strictly required anymore.

But I'd still expect a such array to be ordered by default, to not rely on the JS implementation. Also, there is no native `kstort` filter in Twig that could be used as a workaround.

Reply:
But why would you sort [2 => 2, 1 => 1] for example ?

Reply:
Hum because I tend to expect an array numerically indexed to be ordered. At least from a data source like an URL. But I may miss something here?

Reply:
I can understand why but ... there is nothing saying that in the [URI RFC](https://datatracker.ietf.org/doc/html/rfc3986#section-1.1.1). 

(and that is probably the reason why Symfony\...\Request doesn't do such modification)

And, in fact, the [ ] have no more meaning than "abc" ou "--" ... it's to the backend engine to decide the semantic given to the special chars and how they combine.

The URI (as described in the RFC) must be parsed from left to right. 

PHP found a way to represent array easily, but in the same time lost the `foo=bar&foo=baz` format, that is also a 100% valid URL
(format used by the Web / JS [URL](https://developer.mozilla.org/fr/docs/Web/API/URL) (that you extends from in your uri_utils i think) with its searchUrlParam - almost like how POST data are handled in php) 

-- 

In a pratical point of view, the following arrays are similar, but not equal.

```
$foo = [2 => 2, 0 => 0];
$bar = [0 => 0, 2 => 2];

dump($foo == $bar);    // true
dump($foo === $bar);  // false
dump(serialize($foo) === serialize($bar));  // false
dump(json_encode($foo) === json_encode($bar)); // false
```

Imagine in database i have the array $foo

If i use it as a LiveProp-url-mapped, you will consider it as a different URL / data, that will try to replace the URL... but somehow in the backed i keep my order, and we have an infinite loop

Reply:
Alright I got it, thanks for the explanations! I'll remove this part

Reply:
Oh I'm 100% open to any other views ..  i was just explaining my question / doubts :)

Reply:
I think the RFC argument wins the debate ;) 
My initial point of view was quite arbitrary, soooo let's stick to something solid!